swift-sdk/Internal/IterableActionRunner.swift

```diff
@@ -76,14 +79,12 @@ struct IterableActionRunner {
         }
     }
     
-    // MARK: Private
-    
     private static func shouldOpenUrl(url: URL, from source: IterableActionSource) -> Bool {
         if source == .push || source == .inApp, let scheme = url.scheme, scheme == "http" || scheme == "https" {
             return true
-        } else {
-            return false
         }
+        
+        return false
```

Comment:
Not sure why this is better than what was existing?

Reply:
In general short circuiting is helpful for methods with long bodies.

Reply:
Oh, that was just a clarity change since there's no point in having the `else` there with just a `return` in it.

Reply:
Usually I also don’t like unnecessary else statements containing code. And short circuiting is good. But in this case this does not feel like short circuiting. I like the symmetry of if else. Let’s leave it as it is.

Reply:
I think it still short circuits the same way in both scenarios

Reply:
Of course. Both will do the same thing. I just like the symmetry of this better.

Consider the two scenarios below. For the first, I like return inside else. For the second scenario I like return outside as you propose.

// scenario 1 (currently existing)
if condition1 && condition2 {
   return true
} else {
  return false
}

// scenario 2
if condition1 && condition 2 {
return true
} else {
  // notice more statements
  statement 1
  statement 2
  return false
}

// Scenario 2 should be changed to 
if condition1 && condition 2 {
  return true
}

statement 1
statement 2
return false
}


If you look at Scala code (not that we should follow Scala guidelines for Swift), that is how they write most of their code (without the explicit return).